### PR TITLE
U902-020: Add is_ghost_code property

### PR DESCRIPTION
**Pull request by raph-amiard**

### CODE REVIEW

File: ada/ast.py [hunk 1]
```diff
@@ -7242,6 +7252,15 @@ class Pragma(AdaNode):
 
     xref_entry_point = Property(True)
 
+    @langkit_property(public=True)
+    def is_ghost_code():
+        """
+        Return whether this declaration is ghost code or not. See SPARK RM 6.9.
+        """
+        return Entity.id.sym.any_of(
+            'Assert', 'Assume'
```
Comment (Roldak) [inline ada/ast.py:7261]: Missing `Loop_Invariant`, maybe also `Invariant` (for types), `Pre`, `Post`? (and their 'Class variant?)

File: ada/ast.py [hunk 2]
```diff
@@ -15287,6 +15306,31 @@ class Stmt(AdaNode):
 
     xref_entry_point = Property(True)
 
+    @langkit_property(public=True)
+    def is_ghost_code():
+        """
+        Return whether this declaration is ghost code or not. See SPARK RM 6.9.
```
Comment (Roldak) [inline ada/ast.py:15312]: copy-pasta: `declaration` -> `statement`

**Review by Roldak — CHANGES_REQUESTED** [review]
What about Ada 2012 aspects, shouldn't we also provide `p_is_ghost_code` on them like we have for pragmas? (If so, probably `AdaNode.is_contract_aspect` can help)

**Comment by raph-amiard** [comment]
> What about Ada 2012 aspects, shouldn't we also provide `p_is_ghost_code` on them like we have for pragmas? (If so, probably `AdaNode.is_contract_aspect` can help)

It's not necessary for gnatcov because gnatcov treats aspects differently (since they're never in lists of decls/stmts), hence why I didn't add them

**Review by Roldak — APPROVED** [review]

**Pull request merged**